src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -0,0 +1,1098 @@
+
+            const auto * orc_struct_column = dynamic_cast<const orc::StructVectorBatch *>(orc_column);
+            for (size_t i = 0; i < orc_type->getSubtypeCount(); ++i)
+            {
+                const auto & field_name = orc_type->getFieldName(i);
+
+                DataTypePtr nested_type_hint;
+                if (tuple_type_hint)
+                {
+                    if (tuple_type_hint->haveExplicitNames())
+                    {
+                        auto pos = tuple_type_hint->tryGetPositionByName(field_name);
+                        if (pos)
+                            nested_type_hint = tuple_type_hint->getElement(*pos);
+                    }
+                    else if (size_t(i) < tuple_type_hint->getElements().size())
+                        nested_type_hint = tuple_type_hint->getElement(i);
+                }
+
+                const auto * nested_orc_column = orc_struct_column->fields[i];
+                const auto * nested_orc_type = orc_type->getSubtype(i);
+                auto element = readColumnFromORCColumn(
+                    nested_orc_column,
+                    nested_orc_type,
+                    field_name,
+                    false,
+                    allow_null_type,
+                    skip_columns_with_unsupported_types,
+                    skipped,
+                    nested_type_hint);
+
+                if (skipped)
+                    return {};
+
+                tuple_elements.emplace_back(std::move(element.column));
+                tuple_types.emplace_back(std::move(element.type));
+                tuple_names.emplace_back(std::move(element.name));
+            }
+
+            auto tuple_column = ColumnTuple::create(std::move(tuple_elements));
+            auto tuple_type = std::make_shared<DataTypeTuple>(std::move(tuple_types), std::move(tuple_names));
+            return {std::move(tuple_column), std::move(tuple_type), column_name};
+        }
+        default: {
+            if (skip_columns_with_unsupported_types)
+            {
+                skipped = true;
+                return {};
+            }
+
+            throw Exception(
+                ErrorCodes::UNKNOWN_TYPE,
+                "Unsupported ORC type '{}' of an input column '{}'. "
+                "If it happens during schema inference and you want to skip columns with "
+                "unsupported types, you can enable setting input_format_orc"
+                "_skip_columns_with_unsupported_types_in_schema_inference",
+                orc_type->toString(),
+                column_name);
+        }
+    }
+}
+
+void ORCColumnToCHColumn::orcColumnsToCHChunk(
+    Chunk & res, NameToColumnPtr & name_to_column_ptr, size_t num_rows, BlockMissingValues * block_missing_values)
+{
+    Columns columns_list;
+    columns_list.reserve(header.columns());
+    std::unordered_map<String, std::pair<BlockPtr, std::shared_ptr<NestedColumnExtractHelper>>> nested_tables;
+    bool skipped = false;
```

Comment:
Should be inside the loop? (Or likely shouldn't exist at all, see above.)

Reply:
It should bt kept the same with `ArrowColumnToCHColumn::arrowColumnsToCHChunk`